akka-cluster/src/main/scala/akka/cluster/ClusterHeartbeat.scala

```diff
@@ -249,14 +269,35 @@ private[cluster] case class HeartbeatNodeRing(selfAddress: UniqueAddress, nodes:
   /**
    * The receivers to use from a specified sender.
    */
-  def receivers(sender: UniqueAddress): immutable.Set[UniqueAddress] =
+  def receivers(sender: UniqueAddress): Set[UniqueAddress] =
     if (useAllAsReceivers)
       nodeRing - sender
     else {
-      val slice = nodeRing.from(sender).tail.take(monitoredByNrOfMembers)
-      if (slice.size < monitoredByNrOfMembers)
-        (slice ++ nodeRing.take(monitoredByNrOfMembers - slice.size))
-      else slice
+
+      // Pick nodes from the iterator until n nodes that are not unreachable have been selected.
+      // Intermediate unreachable nodes are also included in the result.
```

Comment:
To summarize my thinking:
if the node has monitored someone from the shutdown nodes, then this forces it to monitor another one, too. Now if that happens to be another shutdown node, we are lucky and extend again. But what if this node has monitored none of the shutdown ones, then no extension happens. Or what if the extension process stops because we pick a replacement node that is still up? Or we assume this is unlikely?